gnucash/register/register-core/table-model.h

```diff
@@ -189,25 +185,14 @@ TableGetCellIOFlagsHandler gnc_table_model_get_io_flags_handler
 (TableModel *model,
  const char * cell_name);
 
-void gnc_table_model_set_fg_color_handler
-(TableModel *model,
- TableGetFGColorHandler io_flags_handler,
- const char * cell_name);
-void gnc_table_model_set_default_fg_color_handler
-(TableModel *model,
- TableGetFGColorHandler io_flags_handler);
-TableGetFGColorHandler gnc_table_model_get_fg_color_handler
-(TableModel *model,
- const char * cell_name);
-
-void gnc_table_model_set_bg_color_handler
+void gnc_table_model_set_cell_color_handler
 (TableModel *model,
- TableGetBGColorHandler io_flags_handler,
+ TableGetCellColorHandler io_flags_handler,
  const char * cell_name);
-void gnc_table_model_set_default_bg_color_handler
+ void gnc_table_model_set_default_cell_color_handler
 (TableModel *model,
- TableGetBGColorHandler io_flags_handler);
-TableGetBGColorHandler gnc_table_model_get_bg_color_handler
+ TableGetCellColorHandler io_flags_handler);
+TableGetCellColorHandler gnc_table_model_get_cell_color_handler
```

Comment:
This commit will break the build until the next commit, because you replace the fg/bg function definitions with a single cell variant, but the rest of the code is still calling the fg/bg functions.

You should try to avoid this as it makes bisecting the git history more difficult. git bisect is a crucial tool to figure out in which commit a bug first appears, but if there are commits with build errors the bisect can't be used accurately.

If you really have a good reason to deliberately insert a commit that breaks the build, this should be explained in the commit message. In this case I don't see a good reason. This and the next commit are functionally part of the same change, or at least the parts in the next commit where you replace the fg/bg functions. I'm not sure if you can or want to keep the ENUM reduction separately.

Other than that, I'm very much in favor of this simplification :)